calcom/cal.com
# feat: Add consistent iCalUID

## What does this PR do?



This is a part of stacked PRs that aim to refactor how we handle `.ics` files.

This PR adds the `iCalUID` and the `iCalSequence` to the booking record. This is a part of a stack PRs that aims to unify the `.ics` file that we sent.

Fixes # (issue)



## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change





## How should this be tested?



- Set GCal as your destination calendar
- Create a booking
    - The `iCalUID` should be `${uid}@cal.com` in the DB and the .ics file sent
    - The `iCalSequence` should be set to 0
- Reschedule the booking
    - In the new booking, the `iCalUID` should be the same as the original booking
    - The `iCalSequence` should be set to 1
- Reschedule that booking a second time
    - The sequence should be set to 2
- Cancel that booking
    - The sequence should be 3
-  Now set Outlook as your destination calendar and create a booking
    - The `iCalUID` should come from Outlook as you can not set the `icalUID` through the API

## Mandatory Tasks


## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
